### ISSUE TITLE
Xournalpp not catch the second pen button click on Last DECO 01v2XP-PEN dirver 

### ISSUE THREAD

**Issue by Alfystar** [xournalpp/xournalpp#3083]
(Please complete the following information, and then delete this line)

**Affects versions :**
 - OS: Linux, kububto 21.04
 - Desktop environment: Xorg
 - Xournal++ version and libgtk do you use
     Xournal++ 1.1.0+dev
      └──libgtk: 3.24.25
 - Installation method:
    direct compile from source 

**Describe the bug**
Xournalpp not catch the second pen button click on Last DECO 01v2XP-PEN dirver 
Only the first button work well

**MODEL Graphic Tablet **
DECO 01v2XP-PEN Deco 01 V2 Tavoletta Grafica 5080

**Comment by rolandlo** [comment]
Please compile via 
```
cmake -DDEBUG_INPUT=ON ..
cmake --build .
```
and watch the input debugging messages while you press the non-working button. Report back your observations. See also [the wiki page](https://github.com/xournalpp/xournalpp/wiki/How-to-provide-debug-info-for-input-related-bugs)

**Comment by Alfystar** [comment]
Do it, this is the result:
```
$ ./src/xournalpp 
** Message: 13:29:50.618: TEXTDOMAINDIR = (null), PACKAGE_LOCALE_DIR = /usr/local/share/locale, chosen directory = /usr/local/share/locale
ALSA lib pcm_dsnoop.c:638:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dmix.c:1075:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_dmix.c:1075:(snd_pcm_dmix_open) unable to open slave
** Message: 13:29:50.733: Plugin "MigrateFontSizes" UI initialized

** (xournalpp:20124): WARNING **: 13:29:50.801: You can use the toolbar item "PEN" only once!

** (xournalpp:20124): WARNING **: 13:29:50.802: You can use the toolbar item "HIGHLIGHTER" only once!

** (xournalpp:20124): WARNING **: 13:29:50.802: You can use the toolbar item "UNDO" only once!

** (xournalpp:20124): WARNING **: 13:29:50.802: You can use the toolbar item "ZOOM_SLIDER" only once!

** (xournalpp:20124): WARNING **: 13:29:50.802: You can use the toolbar item "DRAW_ELLIPSE" only once!

** (xournalpp:20124): WARNING **: 13:29:50.802: You can use the toolbar item "TOOL_FILL" only once!

** (xournalpp:20124): WARNING **: 13:29:50.802: You can use the toolbar item "FINE" only once!

** (xournalpp:20124): WARNING **: 13:29:50.802: You can use the toolbar item "THICK" only once!

** (xournalpp:20124): WARNING **: 13:29:50.802: You can use the toolbar item "DRAW_RECTANGLE" only once!

** (xournalpp:20124): WARNING **: 13:29:50.802: You can use the toolbar item "DRAW_ARROW" only once!
** Message: 13:29:50.896: Event
Event type:     GDK_FOCUS_CHANGE
Source device:  GDK_SOURCE_KEYBOARD
Device Class:   INPUT_DEVICE_KEYBOARD

** Message: 13:29:51.451: Event
Event type:     GDK_ENTER_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:51.453: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:54.154: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 13:29:54.161: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:54.386: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 13:29:54.467: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:55.710: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 13:29:55.717: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:55.723: Event
Event type:     GDK_LEAVE_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:55.725: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:56.317: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 13:29:56.317: Event
Event type:     GDK_LEAVE_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:57.109: Event
Event type:     GDK_ENTER_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:57.121: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:58.204: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 13:29:58.211: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:59.201: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 13:29:59.247: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:59.818: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 13:29:59.825: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:59.840: Event
Event type:     GDK_LEAVE_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:29:59.840: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:00.767: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 13:30:00.767: Event
Event type:     GDK_LEAVE_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:01.916: Event
Event type:     GDK_ENTER_NOTIFY
Source device:  GDK_SOURCE_MOUSE
Device Class:   INPUT_DEVICE_MOUSE

** Message: 13:30:01.959: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:02.496: Event
Event type:     GDK_LEAVE_NOTIFY
Source device:  GDK_SOURCE_MOUSE
Device Class:   INPUT_DEVICE_MOUSE

** Message: 13:30:02.497: Event
Event type:     GDK_ENTER_NOTIFY
Source device:  GDK_SOURCE_MOUSE
Device Class:   INPUT_DEVICE_MOUSE

** Message: 13:30:02.497: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 13:30:02.503: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:02.524: Event
Event type:     GDK_LEAVE_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:02.524: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:03.178: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 13:30:03.178: Event
Event type:     GDK_LEAVE_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:03.451: Event
Event type:     GDK_ENTER_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:03.455: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:04.152: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 13:30:04.159: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:05.256: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 13:30:05.267: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:06.080: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 13:30:06.087: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:06.096: Event
Event type:     GDK_LEAVE_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:06.098: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:07.267: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 13:30:07.267: Event
Event type:     GDK_LEAVE_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:07.729: Event
Event type:     GDK_ENTER_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:07.729: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:08.434: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 13:30:08.444: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:08.915: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 13:30:08.961: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:09.535: Event
Event type:     GDK_LEAVE_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 13:30:10.666: Event
Event type:     GDK_FOCUS_CHANGE
Source device:  GDK_SOURCE_KEYBOARD
Device Class:   INPUT_DEVICE_KEYBOARD

```

The first warning are my personalization, (other pull request)
Hope this is enouth to solve :four_leaf_clover: 

Emanuele

**Comment by rolandlo** [comment]
As I see, the pen button you pressed (right in the beginning and multiple times) is Button 3. In the stylus panel in menu `Edit > Preferences` this corresponds to `button 2`. What tool have you set there?
Button 3 is also used to clone selections. What happens if you select some stroke and press the stylus button?

**Comment by Alfystar** [comment]
First, sorry for this big delay, unfortunaly i catch same unexpected events :sweat:

About the problem here the responde:
> in the stylus panel in menu Edit > Preferences this corresponds to button 2. What tool have you set there?

![image](https://user-images.githubusercontent.com/32840211/116941314-be5c4f00-ac6f-11eb-90da-51e214c36cd2.png)
This are my configuration, this pen model haven't the ereser, so i need to add it no the first button (the only one functioning now)

**Comment by rolandlo** [comment]
Can you please try again with version 1.1.0?

**Comment by Alfystar** [comment]
I tried, but when i recompilig the last version of the repository with the debug flag active: `cmake -DDEBUG_INPUT=ON ..`,  keep out this error during compilation:
```
alfy@Alfy-Asus:~/Documenti/xournalpp/debugBuild(master)$ make 
[  0%] Build translations.
[ 17%] Built target translations
[ 25%] Built target util
[ 26%] Building CXX object src/CMakeFiles/xournalpp-core.dir/gui/inputdevices/InputContext.cpp.o
/home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.cpp: In member function ‘void InputContext::printDebug(GdkEvent*)’:
/home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.cpp:209:5: error: ‘string’ was not declared in this scope
  209 |     string message = "Event\n";
      |     ^~~~~~
/home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.cpp:209:5: note: suggested alternatives:
In file included from /usr/include/c++/9/string:39,
                 from /usr/include/c++/9/stdexcept:39,
                 from /usr/include/c++/9/optional:38,
                 from /usr/include/c++/9/bits/node_handle.h:39,
                 from /usr/include/c++/9/bits/stl_tree.h:72,
                 from /usr/include/c++/9/set:60,
                 from /home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.h:15,
                 from /home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.cpp:5:
/usr/include/c++/9/bits/stringfwd.h:79:33: note:   ‘std::string’
   79 |   typedef basic_string<char>    string;
      |                                 ^~~~~~
In file included from /usr/include/c++/9/stdexcept:39,
                 from /usr/include/c++/9/optional:38,
                 from /usr/include/c++/9/bits/node_handle.h:39,
                 from /usr/include/c++/9/bits/stl_tree.h:72,
                 from /usr/include/c++/9/set:60,
                 from /home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.h:15,
                 from /home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.cpp:5:
/usr/include/c++/9/string:67:11: note:   ‘std::pmr::string’
   67 |     using string    = basic_string<char>;
      |           ^~~~~~
/home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.cpp:210:11: error: expected ‘;’ before ‘gdkEventTypes’
  210 |     string gdkEventTypes[] = {"GDK_NOTHING",
      |           ^~~~~~~~~~~~~~
      |           ;
/home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.cpp:260:5: error: ‘message’ was not declared in this scope; did you mean ‘g_message’?
  260 |     message += "Event type:\t" + gdkEventTypes[event->type + 1] + "\n";
      |     ^~~~~~~
      |     g_message
/home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.cpp:260:34: error: ‘gdkEventTypes’ was not declared in this scope; did you mean ‘GdkEventType’?
  260 |     message += "Event type:\t" + gdkEventTypes[event->type + 1] + "\n";
      |                                  ^~~~~~~~~~~~~
      |                                  GdkEventType
/home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.cpp:262:11: error: expected ‘;’ before ‘gdkInputSources’
  262 |     string gdkInputSources[] = {"GDK_SOURCE_MOUSE",    "GDK_SOURCE_PEN",        "GDK_SOURCE_ERASER",
      |           ^~~~~~~~~~~~~~~~
      |           ;
/home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.cpp:266:37: error: ‘gdkInputSources’ was not declared in this scope; did you mean ‘GdkInputSource’?
  266 |     message += "Source device:\t" + gdkInputSources[gdk_device_get_source(device)] + "\n";
      |                                     ^~~~~~~~~~~~~~~
      |                                     GdkInputSource
/home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.cpp:267:11: error: expected ‘;’ before ‘gdkInputClasses’
  267 |     string gdkInputClasses[] = {"INPUT_DEVICE_MOUSE",    "INPUT_DEVICE_PEN",
      |           ^~~~~~~~~~~~~~~~
      |           ;
/home/alfy/Documenti/xournalpp/src/gui/inputdevices/InputContext.cpp:272:36: error: ‘gdkInputClasses’ was not declared in this scope
  272 |     message += "Device Class:\t" + gdkInputClasses[deviceClass] + "\n";
      |                                    ^~~~~~~~~~~~~~~
make[2]: *** [src/CMakeFiles/xournalpp-core.dir/build.make:1857: src/CMakeFiles/xournalpp-core.dir/gui/inputdevices/InputContext.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:292: src/CMakeFiles/xournalpp-core.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

```

so i can't try, but normal compiling o the final product compile.

So i test the new new driviver (2 july 2021 version) on the 1.1.0 complile, but continue to fail

PS
I test respect the last commit in mainline today:
```
alfy@Alfy-Asus:~/Documenti/xournalpp/debugBuild(master)$ git show
commit 976cdce384b04f0f8aea130b320b6f84620ac3d5 (HEAD -> master, origin/master, origin/HEAD)
Author: Benjamin Hennion <benjamin.hennion@wanadoo.fr>
Date:   Sat Jul 24 22:17:38 2021 +0200

    Add full stroke repaint when using highligher and Fill

diff --git a/src/control/tools/StrokeHandler.cpp b/src/control/tools/StrokeHandler.cpp
index d70a53bb..aa13255a 100644
--- a/src/control/tools/StrokeHandler.cpp
+++ b/src/control/tools/StrokeHandler.cpp
@@ -269,7 +269,7 @@ void StrokeHandler::onButtonReleaseEvent(const PositionInputData& pos) {
     page->fireElementChanged(stroke);
 
     // Manually force the rendering of the stroke, if no motion event occurred between, that would rerender the page.
-    if (stroke->getPointCount() == 2) {
+    if (stroke->getPointCount() == 2 || (stroke->getToolType() == STROKE_TOOL_HIGHLIGHTER && stroke->getFill() != -1)) {
         this->redrawable->rerenderElement(stroke);
     }

```

**Comment by rolandlo** [comment]
That's a regression from #2468. You just need to replace `string`by `std::string` throughout the whole file `InputContext.cpp`. I will fix this in a new PR and check whether there are more such errors hiding behind compile flags.

**Comment by Alfystar** [comment]
fix made, compile success, this is the output:
---
```
** Message: 18:51:50.604: TEXTDOMAINDIR = (null), PACKAGE_LOCALE_DIR = /usr/local/share/locale, chosen directory = /usr/local/share/locale
ALSA lib pcm_dsnoop.c:641:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dmix.c:1089:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm.c:2642:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2642:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2642:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_dmix.c:1089:(snd_pcm_dmix_open) unable to open slave
** Message: 18:51:50.795: Plugin "MigrateFontSizes" UI initialized

** (xournalpp:15375): WARNING **: 18:51:50.855: You can use the toolbar item "PEN" only once!

** (xournalpp:15375): WARNING **: 18:51:50.855: You can use the toolbar item "HIGHLIGHTER" only once!

** (xournalpp:15375): WARNING **: 18:51:50.855: You can use the toolbar item "UNDO" only once!

** (xournalpp:15375): WARNING **: 18:51:50.855: You can use the toolbar item "ZOOM_SLIDER" only once!

** (xournalpp:15375): WARNING **: 18:51:50.855: You can use the toolbar item "DRAW_ELLIPSE" only once!

** (xournalpp:15375): WARNING **: 18:51:50.855: You can use the toolbar item "TOOL_FILL" only once!

** (xournalpp:15375): WARNING **: 18:51:50.855: You can use the toolbar item "FINE" only once!

** (xournalpp:15375): WARNING **: 18:51:50.855: You can use the toolbar item "THICK" only once!

** (xournalpp:15375): WARNING **: 18:51:50.855: You can use the toolbar item "DRAW_RECTANGLE" only once!

** (xournalpp:15375): WARNING **: 18:51:50.856: You can use the toolbar item "DRAW_ARROW" only once!
** Message: 18:51:50.929: Event
Event type:     GDK_FOCUS_CHANGE
Source device:  GDK_SOURCE_KEYBOARD
Device Class:   INPUT_DEVICE_KEYBOARD

** Message: 18:51:50.978: Event
Event type:     GDK_ENTER_NOTIFY
Source device:  GDK_SOURCE_MOUSE
Device Class:   INPUT_DEVICE_MOUSE

** Message: 18:51:52.018: Event
Event type:     GDK_LEAVE_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:51:53.531: Event
Event type:     GDK_ENTER_NOTIFY
Source device:  GDK_SOURCE_MOUSE
Device Class:   INPUT_DEVICE_MOUSE

** Message: 18:51:53.533: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_MOUSE
Device Class:   INPUT_DEVICE_MOUSE

** Message: 18:51:53.634: Event
Event type:     GDK_LEAVE_NOTIFY
Source device:  GDK_SOURCE_MOUSE
Device Class:   INPUT_DEVICE_MOUSE

** Message: 18:51:54.716: Event
Event type:     GDK_ENTER_NOTIFY
Source device:  GDK_SOURCE_MOUSE
Device Class:   INPUT_DEVICE_MOUSE

** Message: 18:51:54.718: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_MOUSE
Device Class:   INPUT_DEVICE_MOUSE

** Message: 18:51:56.037: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 18:51:56.044: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:51:57.326: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 18:51:57.335: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:51:59.622: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 18:51:59.633: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:51:59.812: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 18:51:59.930: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:51:59.932: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 18:51:59.935: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:51:59.936: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 18:51:59.942: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:51:59.997: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 18:51:59.999: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 18:52:00.274: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:00.275: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 18:52:00.279: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:00.279: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 18:52:00.285: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:00.352: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 18:52:00.361: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:00.450: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 18:52:00.540: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:00.814: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 18:52:00.825: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:00.991: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 18:52:00.998: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:01.476: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 18:52:01.480: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:01.537: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 3

** Message: 18:52:01.582: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:02.260: Event
Event type:     GDK_KEY_PRESS
Source device:  GDK_SOURCE_KEYBOARD
Device Class:   INPUT_DEVICE_KEYBOARD

** Message: 18:52:02.262: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:02.286: Event
Event type:     GDK_KEY_RELEASE
Source device:  GDK_SOURCE_KEYBOARD
Device Class:   INPUT_DEVICE_KEYBOARD

** Message: 18:52:02.301: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:02.471: Event
Event type:     GDK_BUTTON_PRESS
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 18:52:02.478: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:03.659: Event
Event type:     GDK_BUTTON_RELEASE
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN
Button: 1

** Message: 18:52:03.674: Event
Event type:     GDK_MOTION_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:04.798: Event
Event type:     GDK_LEAVE_NOTIFY
Source device:  GDK_SOURCE_PEN
Device Class:   INPUT_DEVICE_PEN

** Message: 18:52:06.091: Event
Event type:     GDK_FOCUS_CHANGE
Source device:  GDK_SOURCE_KEYBOARD
Device Class:   INPUT_DEVICE_KEYBOARD
```


Now the event is caputure, this is my current pen setting (from the last driver):
![image](https://user-images.githubusercontent.com/32840211/127028337-306255dc-c0a5-4b20-a8d5-c3a26981266a.png)


Hope can help you :smile:

**Comment by rolandlo** [comment]
Seems the eraser button it is not supported by then X-Pen driver on Linux:

https://www.xp-pen.com/product/461.html
![grafik](https://user-images.githubusercontent.com/40485433/127062815-75c98cdb-df48-4aff-9951-5a54381db0cb.png)

**Comment by Alfystar** [comment]
Is an hardware limitation, the ereser isn't preset, for that reason I use one button for that (the only one button function)
The second one now are detecteded, but I suppose you have to work to connect the signal to the right function, right?

**Comment by Alfystar** [comment]
FINALY, with the Last Deco Driver (XP-PEN-pentablet-3.2.0.210824-1.x86_64.deb, Aug 26,2021 AM 11:10)
The problem are "solve" now eraser and pen is reponsability of the driver, xournalpp see the change like the hw eraser, only the 2° button work

The solution work on the last build config, directly from the source :
Version: 1.1.0
Git com/mit: aca20186
Build: 30/08/21
Libgtk: 3.24.25

How this is forever

**Comment by bhennion** [comment]
So... Is this fixed or is there still an issue?

**Comment by Alfystar** [comment]
> So... Is this fixed or is there still an issue?

We can say yes, is different from the original behaviour, but is safe and work good

Now is more similar to the behaviour in Microsoft word (that's not so great, but I suppose is the original goal)

**Comment by bhennion** [comment]
I don't know if that's the case for your device: some devices do not have a physical eraser tip on the other end. Sometimes, the driver emulates the eraser by linking it to one of the side buttons.

You can test this by
- running `evtest | grep BTN` in a terminal, entering the number corresponding to your pen device and leave it running
- touching the tablet/screen with the stylus tip and pressing the various buttons.

You can then quit `evtest` (e.g. by hitting `CTRL + C`) and post the output here. We'll then know what your driver is capable of and how it interprets the button events.

**Comment by Alfystar** [comment]
With the device connected, this 3 event is see, and the event code are, for each of them, in the followind output:
```
alfy@Alfy-Legion-5-Pro:~$ evtest | grep BTN
No device specified, trying to scan all of /dev/input/event*
Not running as root, no devices may be available.
Available devices:
/dev/input/event31:	UGTABLET 10 inch PenTablet Mouse
/dev/input/event256:	UGTABLET 10 inch PenTablet Keyboard
/dev/input/event257:	UGTABLET 10 inch PenTablet
Select the device event number [0-257]: 31
    Event code 272 (BTN_LEFT)
    Event code 273 (BTN_RIGHT)
    Event code 274 (BTN_MIDDLE)
    Event code 275 (BTN_SIDE)
    Event code 276 (BTN_EXTRA)
    Event code 330 (BTN_TOUCH)
^C
alfy@Alfy-Legion-5-Pro:~$ evtest | grep BTN
No device specified, trying to scan all of /dev/input/event*
Not running as root, no devices may be available.
Available devices:
/dev/input/event31:	UGTABLET 10 inch PenTablet Mouse
/dev/input/event256:	UGTABLET 10 inch PenTablet Keyboard
/dev/input/event257:	UGTABLET 10 inch PenTablet
Select the device event number [0-257]: 256
^C
alfy@Alfy-Legion-5-Pro:~$ evtest | grep BTN
No device specified, trying to scan all of /dev/input/event*
Not running as root, no devices may be available.
Available devices:
/dev/input/event31:	UGTABLET 10 inch PenTablet Mouse
/dev/input/event256:	UGTABLET 10 inch PenTablet Keyboard
/dev/input/event257:	UGTABLET 10 inch PenTablet
Select the device event number [0-257]: 257
    Event code 320 (BTN_TOOL_PEN)
    Event code 330 (BTN_TOUCH)
    Event code 331 (BTN_STYLUS)
^C
```

I hope that could be help you

**Comment by bhennion** [comment]
The relevant output here is the last lines:
```
    Event code 320 (BTN_TOOL_PEN)
    Event code 330 (BTN_TOUCH)
    Event code 331 (BTN_STYLUS)
```
It tells you your device has (as far as your driver is concerned) only one button `BTN_TOOL_PEN`. `BTN_STYLUS` corresponds to the stylus hovering, `BTN_TOUCH` corresponds to the stylus touching the tablet. 

There is nothing we can do about this on our side. Sorry.
I'm closing this issue.

**Comment by Alfystar** [comment]
Don't worry, during this time deco developed many driver to align the behaviour as the window standard

From now I only have to adapt my self on the new system 😉😉
Thanks for all your help!
